internal/kldcontracts/smartcontractgw.go

```diff
@@ -1185,6 +1211,31 @@ func (g *smartContractGW) addABI(res http.ResponseWriter, req *http.Request, par
 	json.NewEncoder(res).Encode(info)
 }
 
+func (g *smartContractGW) parseBytecode(form url.Values) ([]byte, error) {
+	v := form["bytecode"]
+	if len(v) > 0 {
+		b := strings.TrimLeft(v[0], "0x")
+		return hex.DecodeString(b)
+	} else {
+		return nil, nil
+	}
+}
+
+func (g *smartContractGW) parseABI(form url.Values) (kldbind.ABIMarshaling, error) {
+	v := form["abi"]
+	if len(v) > 0 {
+		a := v[0]
+		var abi kldbind.ABIMarshaling
+		err := json.Unmarshal([]byte(a), &abi)
+		if err != nil {
+			log.Error(err.Error())
```

Comment:
Should there be some context in this log statement?
```suggestion
			log.Errorf("Failed to parse the ABI supplied in the form: %s", err.Error())
```